client/packages/common/src/ui/forms/JsonForms/useDocumentLoader.ts

```diff
@@ -0,0 +1,90 @@
+import { DocumentRegistry, JsonData } from './JsonForm';
+import { useDocument } from './api';
+import { DocumentRegistryFragment } from './api/operations.generated';
+import { useEffect, useState } from 'react';
+
+/**
+ * Information required to create a new document
+ */
+export interface CreateDocument {
+  data: JsonData;
+  documentRegistry: DocumentRegistryFragment;
+}
+
+export interface DocumentDataResponse {
+  isLoading: boolean;
+  error?: string;
+  documentId?: string;
+  data?: JsonData;
+  documentRegistry?: DocumentRegistry;
+}
+
+// eslint-disable-next-line valid-jsdoc
+/**
+ * Manages the document data for a JSON form.
+ * Data is taken either:
+ * - from an api call (if the document already exist)
+ * - from the createDoc (if the document is going to be created)
+ */
+export const useDocumentLoader = (
+  docName: string | undefined,
+  createDoc?: CreateDocument
+): DocumentDataResponse => {
+  const [error, setError] = useState<string | undefined>();
+  const [data, setData] = useState<JsonData | undefined>();
+  // the current document id (undefined if its a new document)
+  const [documentId, setDocumentId] = useState<string | undefined>();
+  const [documentRegistry, setDocumentRegistry] = useState<DocumentRegistry>({
+    formSchemaId: '',
+    jsonSchema: {},
+    uiSchema: { type: 'VerticalLayout' },
+  });
+
+  // fetch document (only if there is as document name)
+  const {
+    data: databaseResponse,
+    isLoading,
+    isError,
+  } = useDocument.get.document(docName ?? '', !!docName);
+
+  if (isError) {
+    setError(`Failed to load document ${docName}`);
+  }
+  useEffect(() => {
+    if (!databaseResponse) return;
+
+    const { data, documentRegistry } = databaseResponse;
+    if (!data) {
+      setError('No document data');
+    } else {
+      setData(data);
+      setDocumentId(databaseResponse.id);
+    }
+
+    if (!documentRegistry) {
+      setError('No document registry entry');
```

Comment:
are the errors shown to the user? If so, they should be translated

Reply:
Currently not, think the error handling needs to be improved, will leave it for another PR